### DISCUSSION TITLE
Bad Pressure Transducer. Freeze damage or just amazon junk?

### DISCUSSION THREAD

**Discussion by jamestalmage** [tagyoureit/nodejs-poolController#931] [Nixie] [answered]
I bought [this pressure transducer](https://www.amazon.com/gp/product/B07331N8H7/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1).

24v supply, so wires in easily. It worked well last season.

Upon opening this season, it isn't working. It shows 1.4 volts even when the pump is off and the pressure release valve on the top of the filter is open. Voltage doesn't seem to move much as pressure increases. Just seems broken.

Only thing I can think of is that I did not bring it in for the winter. Will freezing break these? (In general, not asking specifically about the random one I selected off Amazon). Rated operating temp is -40˚, so I doubt it.

I am going to inspect my wiring to make sure there hasn't been any water-incursion / shorting during the winter months, but I really doubt it's that.

**Comment by tagyoureit** [comment]
Hard to say.  If it doesn't get to -40 by you, then that probably isn't it.  If you need to replace it, the 5v ones are sufficient and less expensive the the 24v counterparts.  Hope you find something with the wiring.

**Reply by jamestalmage** [reply]
Where do you get the 5v supply? A dedicated supply? Off one of the Pi pins? If from the pi, what added protections are necessary for a wire exiting the enclosure?

**Reply by jamestalmage** [reply]
Also, what pressure range do you usually look for? Do you oversize (200psi) so water-hammer, priming, etc, never causes it to exceed the rated pressure, even for an instant? Or do you just try to cover the actual expected filter-pressure range?

What sort of resolution are these capable of with only 5v?

**Reply by tagyoureit** [reply, marked as answer]
You can get 5v from the Pi pinouts directly or from an external power supply.

I had a 100 psi transducer.  I hardly, if ever, saw the psi climb above 40, and tried to keep it under 20 for normal operation, but maybe your pool is different.  If you are getting water-hammer or extremely high priming that would be a concern for the overall plumbing.  

The resolution can still go down to the hundreths, but that's really way beyond what's necessary.  I'd personally be happy with 0.5 psi min resolution.  I don't want to worry about the pool filter getting 0.03 psi more dirty before I need clean it.

**Reply by raidflex** [reply]
> You can get 5v from the Pi pinouts directly or from an external power supply.
> 
> I had a 100 psi transducer. I hardly, if ever, saw the psi climb above 40, and tried to keep it under 20 for normal operation, but maybe your pool is different. If you are getting water-hammer or extremely high priming that would be a concern for the overall plumbing.
> 
> The resolution can still go down to the hundreths, but that's really way beyond what's necessary. I'd personally be happy with 0.5 psi min resolution. I don't want to worry about the pool filter getting 0.03 psi more dirty before I need clean it.

I just recently installed [this pressure transducer](https://www.amazon.com/dp/B0CFJ7BN3L?psc=1&ref=ppx_yo2ov_dt_b_product_details) from Amazon and the volage is at 1.5v at 0psi.  Its supposed to have a range of .5V at 0psi to 4.5V 100psi.

I am using the 5V from the Building Automation V4 8-Layer Stackable HAT. At only 10PSI the voltage jumps to 2.6V, which also seems high for the given PSI, considering it has a range of 0-100psi and 0.5-4.5V.  Is it possible I received a defective transducer or do I just need to force an offset to compensate?

**Reply by tagyoureit** [reply]
If you are getting 1.5v at 0psi I'd double check the wiring or get a new unit.  Anytime you need to start "compensating" for something that should otherwise just work it's a big red flag 🚩.

**Reply by raidflex** [reply]
> If you are getting 1.5v at 0psi I'd double check the wiring or get a new unit. Anytime you need to start "compensating" for something that should otherwise just work it's a big red flag 🚩.

I just tested the 5V terminal and it is in fact outputting 5V. 
Then  I measured the voltage right at Building Automation board terminals and at 0PSI its .450v, which is correct or at least pretty close to .5v.  As the pressure increased the voltages also was correctly increasing linearly.

So for some reason the board is reading about 1.2V higher then the actual voltage. 

I also tried a different 0-10v terminal on the BA board and received the same result.

**Reply by tagyoureit** [reply]
I'd triple check the dip switches.  Sometimes they are backwards from what you think or not aligned intuitively and can give you readings that appear off if they are not set correctly.

**Reply by raidflex** [reply]
> I'd triple check the dip switches. Sometimes they are backwards from what you think or not aligned intuitively and can give you readings that appear off if they are not set correctly.

Wow I feel stupid lol. I had the wrong dip switch set for a different terminal. The 3rd terminal's dip switches are above the connectors and not the dip switches right next to inputs 1 and 2.
